python/dgl/graphbolt/impl/fused_csc_sampling_graph.py

```diff
@@ -615,21 +658,27 @@ def sample_neighbors(
                     indices=tensor([2]),
         )}
         """
-        if isinstance(nodes, dict):
-            nodes = self._convert_to_homogeneous_nodes(nodes)
-
         return_eids = (
             self.edge_attributes is not None
             and ORIGINAL_EDGE_ID in self.edge_attributes
         )
+
+        node_offsets = None
+        if isinstance(nodes, dict):
+            nodes, node_offsets = self._convert_to_homogeneous_nodes(nodes)
+        elif nodes is None and hasattr(self, "_node_type_offset_local_list"):
+            node_offsets = self._node_type_offset_local_list
```

Comment:
1. In which case `node is None`?
2. This is a little bit hacky, where does the _node_type_offset_local_list get initiated?

Reply:
It gets initiated in `SampleNeighbors.py` file. The problem is that we construct a FusedCscSamplingGraph there and flatten the nodes there too by calling `_convert_to_homogeneous_nodes`, however, there is no way to store this object in there.